packages/store/addon/-private/system/backburner.js

```diff
@@ -8,11 +8,13 @@ import Ember from 'ember';
 
 // TODO: expose Ember._Backburner as `import { _Backburner } from '@ember/runloop'` in ember-rfc176-data + emberjs/ember.js
 /*
-  flushRelationships is used by the Graph to batch updates
   syncRelationships is used by the UI to grab updates from the graph
   and update the ManyArrays.
+
+  We may be able to remove this once the new relationship layer is
+  complete.
```

Comment:
Related to the other question, what will allow it's removal?

Reply:
Our scheduler right now does three things (which is why there are three queues)

- it batches payloads that have been pushed for a single flush, thereby allowing us to order the pushes by type (more efficient to push hasMany first. Removing this batching regressed the update scenario by 500%). It's unclear how we'd do this without some form of run loop concept, however we could probably make a simpler queue thing and drop our need for backburner if this was all that was left.
- it batches (and minimizes) updates to hasMany local state in a second pass, this allows us to update the local state with those changes only once instead of potentially many times. This optimization would go away if we were to compute localState lazily on access.
- it notifies the UI layer only once all state has settled (and minimizes those notifications to only one per relationship). We can possibly do away with this once we no longer support sync-observers if state is also lazily calculated.